tlatools/org.lamport.tlatools/src/pcal/trans.java

```diff
@@ -1327,27 +1325,18 @@ private static List<String> fileToStringVector(String fileName) throws FileToStr
     * element is a line of the file.                                       *
     ***********************************************************************/
     {
-        final List<String> inputVec = new ArrayList<>(100);
-    	// TODO use Apache Commons for this
-        try (final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(new FileInputStream(fileName)))) {
-            String nextLine = bufferedReader.readLine();
-            while (nextLine != null) {
-                inputVec.add(nextLine);
-                nextLine = bufferedReader.readLine();
-            }
-        } catch (FileNotFoundException e) {
-            /**************************************************************
-            * Input file could not be found.                              *
-            **************************************************************/
+        Path inputFilePath = Path.of(fileName);
+        if (!Files.exists(inputFilePath)) {
             throw new FileToStringVectorException("Input file " + fileName + " not found.");
+        }
+        try {
+            return Files.readAllLines(inputFilePath);
```

Comment:
Small nit:

Although most programmers consider it uglier, it is generally more correct _and_ more efficient to catch `NoSuchFileException` instead of checking `Files.exists` first. (Ask forgiveness, not permission.) The removed code followed that pattern correctly, although since you switched to NIO you should replace the old and underspecified `FileNotFoundException` with the new and well-defined `NoSuchFileException`.

A bit more detail:
 - When done in two separate steps, the check for existence and the open call do not happen atomically. The filesystem is a large widely-used global shared mutable concurrent variable, and if it is changing while this program runs, then the call to `Files.readAllLines` can end up throwing `NoSuchFileException` despite the existence check. Fortunately on all mainstream filesystems, the check for existence and opening the file are done atomically at open time. (I don't expect this race to be common, but it's easy to handle so we might as well.)
 - When a file is opened for reading, the operating system has to determine whether it exists or not, so checking for existence and then opening essentially doubles the number of filesystem tree descents. Although we don't think about them much, those tree descents can be the dominant cost for reading small files.

Reply:
Good point. Changed this and also removed a place where file existence was checked during command line arg parsing.